### PR TITLE
feat: add vanilla and react configs

### PR DESCRIPTION
**Pull request by sztok7**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -62,13 +60,11 @@ module.exports = {
         "import/no-extraneous-dependencies": "error",
         "import/first": "error",
         "import/no-duplicates": "error",
-        "import/order": "error",
```
Comment (sztok7) [inline index.js:65]: double rule

File: index.js [hunk 2]
```diff
@@ -3,8 +3,6 @@ module.exports = {
         node: true,
         es6: true
     },
-    rules: {
```
Comment (sztok7) [inline index.js:6]: double `rules` section

File: frontend-vanilla.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+                "checksVoidReturn": {
+                    //detects false positives
+                    properties: false
+                }
+            }
+        ],
+        "@typescript-eslint/explicit-function-return-type": ["error", {
+            "allowExpressions": true
+        }],
+        "@typescript-eslint/consistent-type-definitions": ["error", "type"],
```
Comment (sztok7) [inline frontend-vanilla.js:153]: I think it's a good practice to use either types or interfaces (personally prefer types, but this rule can be changed or turned off if preferences differ or if the project is a dependency of another that relies on a specific type definition style)

File: src/index.spec.ts [hunk 4]
```diff
@@ -2,6 +2,6 @@ import { expect } from "chai";
 
 describe("some test", function () {
   it("test", function () {
-    expect(true).to.be.true;
+    expect(true).to.be.equal(true);
```
Comment (sztok7) [inline src/index.spec.ts:5]: got a warning that it's better to use this 🤷

File: frontend-vanilla.js (hunk 3 above)
Comment (mpetrunic) [reply frontend-vanilla.js:153]: 🤷🏻‍♂️ I kinda use interface for complex types and types when I need one line wrappers

File: src/index.ts [hunk 5]
```diff
@@ -1,12 +1,22 @@
 import { MochaInstanceOptions } from "mocha";
+import Plus from '@/assets/plus.svg';
+import { Button } from '@/components/common/Button/Button';
+import type { SwapAssetExt } from '@/root/trade/_components/TradeView/types';
+import { PREVIEW_TYPES } from '@/services/contexts/preview/constants';
+import { useItemPreviewContext } from '@/services/contexts/preview/ItemPreviewContext';
+import type { SwapAsset } from '@/services/hooks/use-filter-swaps/types';
+import styles from './trade-view-item.module.scss';
```
Comment (mpetrunic) [inline src/index.ts]: how about separate file and test file for frontend configs? :)

Comment (sztok7) [reply src/index.ts]: makes sense

**Review by mpetrunic — APPROVED** [review]

**Pull request merged**